converter/src/main/java/org/dswarm/converter/schema/SolrSchemaParser.java

```diff
@@ -0,0 +1,372 @@
+import org.dswarm.persistence.model.schema.proxy.ProxySchema;
+import org.dswarm.persistence.model.schema.utils.SchemaUtils;
+import org.dswarm.persistence.service.UUIDService;
+import org.dswarm.persistence.service.schema.AttributePathService;
+import org.dswarm.persistence.service.schema.AttributeService;
+import org.dswarm.persistence.service.schema.ClaszService;
+import org.dswarm.persistence.service.schema.SchemaService;
+
+/**
+ * Transforms a given Solr schema file (schema.xml) to a d:swarm schema.
+ *
+ * @author tgaengler
+ */
+public class SolrSchemaParser {
+
+	private static final Logger LOG = LoggerFactory.getLogger(SolrSchemaParser.class);
+
+	private final Provider<SchemaService> schemaServiceProvider;
+
+	private final Provider<ClaszService> classServiceProvider;
+
+	private final Provider<AttributePathService> attributePathServiceProvider;
+
+	private final Provider<AttributeService> attributeServiceProvider;
+
+	private static final String SCHEMA_IDENTIFIER               = "schema";
+	private static final String FIELDS_IDENTIFIER               = "fields";
+	private static final String FIELD_IDENTIFIER                = "field";
+	private static final String FIELDS_XPATH_EXPRESSION         =
+			SchemaUtils.SLASH + SCHEMA_IDENTIFIER + SchemaUtils.SLASH + FIELDS_IDENTIFIER + SchemaUtils.SLASH + FIELD_IDENTIFIER;
+	private static final String NAME_IDENTIFIER                 = "name";
+	private static final String DEFAULT_RECORD_CLASS_LOCAL_NAME = "RecordType";
+
+	private static final DocumentBuilderFactory documentBuilderFactory;
+	private static final XPathFactory           xPathfactory;
+
+	static {
+
+		documentBuilderFactory = DocumentBuilderFactory.newInstance();
+		xPathfactory = XPathFactory.newInstance();
+	}
+
+	@Inject
+	public SolrSchemaParser(final Provider<SchemaService> schemaServiceProviderArg,
+			final Provider<ClaszService> classServiceProviderArg, final Provider<AttributePathService> attributePathServiceProviderArg,
+			final Provider<AttributeService> attributeServiceProviderArg) {
+
+		schemaServiceProvider = schemaServiceProviderArg;
+		classServiceProvider = classServiceProviderArg;
+		attributePathServiceProvider = attributePathServiceProviderArg;
+		attributeServiceProvider = attributeServiceProviderArg;
+	}
+
+	public Optional<Schema> parse(final String solrSchemaFilePath, final String schemaUUID, final String schemaName)
+			throws DMPPersistenceException {
+
+		final Optional<Document> optionalDocument = readXML(solrSchemaFilePath);
+
+		if (!optionalDocument.isPresent()) {
+
+			LOG.error("parsed Solr schema (from '" + solrSchemaFilePath + "') is not present");
+
+			return Optional.empty();
+		}
+
+		final Document document = optionalDocument.get();
+
+		final Optional<NodeList> optionalFields = getFields(document, solrSchemaFilePath);
+
+		if (!optionalFields.isPresent()) {
+
+			LOG.error("couldn't find fields in the Solr schema (from '" + solrSchemaFilePath + "')");
```

Comment:
`LOG.error("couldn't find fields in the Solr schema (from '{}')", solrSchemaFilePath);`